modules/dnn/src/layers/batch_norm_layer.cpp

```diff
@@ -100,15 +100,10 @@ class BatchNormLayerImpl CV_FINAL : public BatchNormLayer
             dstWeightsData[i] = w;
             dstBiasData[i] = (hasBias ? biasData[i] : 0.0f) - w * meanData[i] * varMeanScale;
         }
-        // We will use blobs to store origin weights and bias to restore them in case of reinitialization.
-        weights_.copyTo(blobs[0].reshape(1, 1));
-        bias_.copyTo(blobs[1].reshape(1, 1));
     }
 
     virtual void finalize(InputArrayOfArrays, OutputArrayOfArrays) CV_OVERRIDE
     {
-        blobs[0].reshape(1, 1).copyTo(weights_);
-        blobs[1].reshape(1, 1).copyTo(bias_);
```

Comment:
This change causes the BatchNorm test to fail. It makes sense as the test was added for this exact change #17284.

In a quantized net, if the int8 version of a layer is not present (like BatchNorm), the fp32 version of the layer is copied along with its params. But here we are overwriting the layer's original parameters with fused ones which causes the layer in the quantized net to give wrong results (NaNs and Infs). Is there a way to solve both the issues?

Reply:
@SamFC10, the compatibility is the first priority, so we need to find some method to provide the information that you need while preserving compatibility. For example, prior to copying blobs to weights_ and bias_ we can save the previous state to some other internal arrays.

Reply:
> prior to copying blobs to weights_ and bias_ we can save the previous state to some other internal arrays

Done. Now batchnorm fusion test pass and quantized nets with fp32 batchnorm layer also work.